pychunkedgraph/backend/chunkedgraph.py

```diff
@@ -1276,91 +1316,91 @@ def add_atomic_edges_in_chunks(self, edge_id_dict: dict,
-                disconnected_areas = np.concatenate([
-                    edge_area_dict["in_disconnected"][ec1_ids],
-                    edge_area_dict["in_disconnected"][ec2_ids]])
+                    parent_cross_edges = np.concatenate([parent_cross_edges, edge_id_dict["between_connected"][row_ids]])
 
+                    time_dict["out_connected"].append(time.time() - time_start_2)
+                    time_start_2 = time.time()
 
                 # out chunk + disconnected
-                ec_mask = edge_id_dict["between_disconnected"][:, 0] == node_id
-                ec_ids = np.where(ec_mask)
+                if node_id in remapping["between_disconnected"]:
+                    row_ids, column_ids = sparse_indices["between_disconnected"][remapping["between_disconnected"][node_id]]
+
+                    row_ids = row_ids[column_ids == 0]
+                    column_ids = column_ids[column_ids == 0]
+                    inv_column_ids = (column_ids + 1) % 2
+                    time_dict["out_disconnected_mask"].append(time.time() - time_start_2)
+                    time_start_2 = time.time()
+
+                    connected_ids = np.concatenate([connected_ids, edge_id_dict["between_disconnected"][row_ids, inv_column_ids]])
+                    connected_affs = np.concatenate([connected_affs, edge_aff_dict["between_disconnected"][row_ids]])
+                    connected_areas = np.concatenate([connected_areas, edge_area_dict["between_disconnected"][row_ids]])
 
-                disconnected_ids = np.concatenate([
-                    disconnected_ids,
-                    edge_id_dict["between_disconnected"][:, 1][ec_ids]])
-                disconnected_affs = np.concatenate([
-                    disconnected_affs,
-                    edge_aff_dict["between_disconnected"][ec_ids]])
-                disconnected_areas = np.concatenate([
-                    disconnected_areas,
-                    edge_area_dict["between_disconnected"][ec_ids]])
+                    time_dict["out_disconnected"].append(time.time() - time_start_2)
+                    time_start_2 = time.time()
 
+                # cross
+                if node_id in remapping["cross"]:
+                    row_ids, column_ids = sparse_indices["cross"][remapping["cross"][node_id]]
+
+                    row_ids = row_ids[column_ids == 0]
+                    column_ids = column_ids[column_ids == 0]
+                    inv_column_ids = (column_ids + 1) % 2
+                    time_dict["cross_mask"].append(time.time() - time_start_2)
+                    time_start_2 = time.time()
+
+                    connected_ids = np.concatenate([connected_ids, edge_id_dict["cross"][row_ids, inv_column_ids]])
+                    connected_affs = np.concatenate([connected_affs, np.full((len(row_ids)), np.inf, dtype=np.float32)])
+                    connected_areas = np.concatenate([connected_areas, np.full((len(row_ids)), np.inf, dtype=np.uint64)])
```

Comment:
`np.inf` can not be represented by a `np.uint64`. Numpy is silently doing an unsafe cast [here](https://github.com/numpy/numpy/blob/v1.15.1/numpy/core/numeric.py#L310) and you miraculously end up with `0`. But that's not something you can/should rely on.